### DISCUSSION TITLE
Selector Precedence

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#10429] [Q&A] [answered]
Considering following snippet:

```xaml
	<Window.Styles>
		<Style Selector="Button[HorizontalAlignment=Left]">
			<Setter Property="Background" Value="Red"/>
		</Style>
		<Style Selector="Button#foo">
			<Setter Property="Background" Value="Green"/>
		</Style>
	</Window.Styles>

	<StackPanel>
		<Button Content="Ok"/>
		<Button x:Name="foo" Content="Foo" IsDefault="True"/>
	</StackPanel>
```
![immagine](https://user-images.githubusercontent.com/12531229/220597183-8d7c9128-06a0-4ec2-bde4-9fbbb2650f4c.png)

I expected Button foo to be Green but it's red.

Why?

**Comment by maxkatz6** [comment, marked as answer]
Similar to this one: https://docs.avaloniaui.net/docs/styling/troubleshooting#missing-style-pseudoclass-trigger-selector

In short, property selector is a trigger selector and has higher priority.
Control name selector is static, so it's not a trigger selector, it has a lower priority.
As per https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Data/BindingPriority.cs#L28
